tests/test_all_fields.py

```diff
@@ -65,65 +62,9 @@ def all_fields_test(self):
-        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
-
-        found_catalogs = menagerie.get_catalogs(conn_id)
-        self.assertGreater(len(found_catalogs), 0, msg="unable to locate schemas for connection {}".format(conn_id))
-
-        found_catalog_names = set(map(lambda c: c['tap_stream_id'], found_catalogs))
-        diff = self.expected_check_streams().symmetric_difference( found_catalog_names )
-        self.assertEqual(0, len(diff), msg="discovered schemas do not match: {}".format(diff))
-        print("discovered schemas are OK")
-
-        # Select all available fields from all testable streams
-        exclude_streams = self.expected_streams().difference(self.TESTABLE_STREAMS)
-        self.select_all_streams_and_fields(
-            conn_id=conn_id, catalogs=found_catalogs, select_all_fields=True, exclude_streams=exclude_streams
-        )
-
-        catalogs = menagerie.get_catalogs(conn_id)
-
-        # Ensure our selection worked
-        for cat in catalogs:
-            catalog_entry = menagerie.get_annotated_schema(conn_id, cat['stream_id'])
-            # Verify all testable streams are selected
-            selected = catalog_entry.get('annotated-schema').get('selected')
-            print("Validating selection on {}: {}".format(cat['stream_name'], selected))
-            if cat['stream_name'] not in self.TESTABLE_STREAMS:
-                self.assertFalse(selected, msg="Stream selected, but not testable.")
-                continue # Skip remaining assertions if we aren't selecting this stream
-            self.assertTrue(selected, msg="Stream not selected.")
-
-            # Verify all fields within each selected stream are selected
-            for field, field_props in catalog_entry.get('annotated-schema').get('properties').items():
-                field_selected = field_props.get('selected')
-                print("\tValidating selection on {}.{}: {}".format(cat['stream_name'], field, field_selected))
-                self.assertTrue(field_selected, msg="Field not selected.")
-
-        #clear state
-        version = menagerie.get_state_version(conn_id)
-        menagerie.set_state(conn_id, {}, version)
-
-        # run sync
-        sync_job_name = runner.run_sync_mode(self, conn_id)
-
-        # Verify tap exit codes
-        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
-        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
-
-        # read target output
-        first_record_count_by_stream = runner.examine_target_output_file(self, conn_id,
-                                                                         self.expected_streams(),
-                                                                         self.expected_primary_keys())
-        replicated_row_count =  sum(first_record_count_by_stream.values())
+        (_, first_record_count_by_stream) = self.run_initial_sync(environment, data_type)
```

Comment:
I'm not sure about pulling all that code from the test. I think it should go back in.